runtime/lua/vim/filetype/detect.lua

```diff
@@ -1459,6 +1459,7 @@ local patterns_hashbang = {
   ['gforth\\>'] = { 'forth', { vim_regex = true } },
   ['icon\\>'] = { 'icon', { vim_regex = true } },
   guile = 'scheme',
+  ['nix-shell'] = 'nix',
```

Comment:
```suggestion
  ['nix%-shell'] = 'nix',
```

I think this also needs that `s:script_checks` change?

Reply:
keep forgetting that these are patterns...

locally this seems to suffice!

Reply:
Btw this is the change that's missing: https://github.com/vim/vim/commit/19548c6a742d954ecd0b50b0680c37cc6ced7473#diff-b8c15b633705ec2abf421ba8d16e5691fea9d4ae34318ac1d80f5f4b8fc4e358R855

Reply:
Wait, I didn't see that before. Added!